### PR TITLE
alphasort location tree issue #124

### PR DESCRIPTION
**Pull request by hiliang-cmu**

### CODE REVIEW

File: src/app/components/locationtree/tree.component.ts [hunk 1]
```diff
@@ -65,9 +65,7 @@ export class TreeNodeComponent {
         .subscribe(
           result => {
             this.childLocations.push(result);
-            if (i === children.length - 1) {
-              this.childLocations.sort(this.compareChildren);
-            }
+            this.childLocations.sort(this.compareChildren);
```
Comment (linux4life798) [inline src/app/components/locationtree/tree.component.ts]: Should probably note (maybe in code) the runtime complexity/performance hit that this fix incurs, for future optimization.

**Comment by linux4life798** [comment]
Nice!
Is this ready to deploy?

Minor Note:
It looks like the insertion and repeated sorting would incur a "worst-case" complexity close to O(n^2*log(n)).
You might want to note (maybe in code) the runtime complexity hit of this fix, for future optimization. This might be something that the REST interface could be help with.

**Comment by hiliang-cmu** [comment]
Minor change to sort only last insertion, accommodates async out-of-order responses. 

Ideal future optimization would indeed be REST interface enhancement, perhaps handling array of 'locations' to consolidate n separate 'location' requests (which I imagine incurs greatest overhead)

**Comment by linux4life798** [comment]
Alright cool

**Pull request merged**